### ISSUE TITLE
Search results sort order

### ISSUE THREAD

**Issue by ksere** [woocommerce/woocommerce#14292]
### EXPLANATION OF THE ISSUE

When searching for a product the search results are sorted alphabetically (and by menu_order) as seen in the main query:

```
...
ORDER BY wp_posts.menu_order ASC, wp_posts.post_title ASC
LIMIT 0, 10
...
```

Now let's say we are searching for a product named 'Woo Ninja', and there is a product with that exact title, and there is also a product named 'Awesome Ninja'. If the 'Awesome Ninja' product has both the terms 'woo' and 'ninja' in it's post_content, it will appear before the 'Woo Ninja' product that has an exact title match.

On a store with a lot of products this behaviour caused the exact match titled product to be pushed to the second page of the search results, instead of appearing 1st.

While debugging i disabled WC and run the search again (after registering manually the 'product' post type). In that case the results returned with the 'proper' order, exact match was first.

### STEPS TO REPRODUCE THE ISSUE

Create a product named 'Woo Ninja'
Create a product named 'Awesome Ninja'
Add the words 'woo' and 'ninja' in the 'Awesome Ninja' product's content.
Search for 'Woo Ninja'.

### SYSTEM STATUS

<details>
`
### WordPress Environment ###

Home URL: http://dev/testsite
Site URL: http://dev/testsite
WC Version: 2.6.14
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_AU

### Server Environment ###

Server Info: Apache/2.4.18 (Ubuntu)
PHP Version: 7.0.15-0ubuntu0.16.04.4
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: N/A
SUHOSIN Installed: –
MySQL Version: 5.7.16
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.6.14
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Active Plugins (2) ###

Query Monitor: by John Blackbourn – 2.13.4
WooCommerce: by WooThemes – 2.6.14

### Settings ###

Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔

### WC Pages ###

Shop Base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My Account: #9 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

### Theme ###

Name: Twenty Sixteen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
See https://github.com/woocommerce/woocommerce/pull/14294 Won't be fixed in 3.0

**Comment by ksere** [comment]
Thanks!
